### PR TITLE
feat(gatsby-source-wordpress): Add retries to graphql requests

### PR DESCRIPTION
**Pull request by TylerBarnes**
This PR adds 5 retries to all WordPress GraphQL requests that fail with a 500, 502, 503, or 504 error.

### CODE REVIEW

File: packages/gatsby-source-wordpress/src/utils/fetch-graphql.ts [hunk 1]
```diff
@@ -27,6 +28,13 @@ export const moduleHelpers = {
   },
 }
 
+const errorIs500ish = (e: Error): boolean =>
+  e.message.includes(`Request failed with status code 50`) &&
+  (e.message.includes(`502`) ||
```
Comment (veryspry) [inline packages/gatsby-source-wordpress/src/utils/fetch-graphql.ts:33]:
```suggestion
  [`502`, `503`, `504`].includes(e.message)
  (e.message.includes(`502`) ||
```

File: packages/gatsby-source-wordpress/src/utils/fetch-graphql.ts [hunk 2]
```diff
@@ -729,9 +732,24 @@ const fetchGraphql = async ({
...
+              // for any error that is not a 50x error, we bail, meaning we stop retrying. error will be thrown one level higher
+              bail(e)
+
+              return null
+            } else {
+              // otherwise throwing the error will cause the retry to happen again
+              throw e
+            }
+          }),
+      { retries: 5 }
```
Comment (veryspry) [inline packages/gatsby-source-wordpress/src/utils/fetch-graphql.ts:751]: Is it worth making this configurable via plugin options?

Comment (TylerBarnes) [reply packages/gatsby-source-wordpress/src/utils/fetch-graphql.ts:751]: The plugin is super bloated with options so I think we should leave it out for now and add it later if needed. It's very easy to add and hard to take away 😅

**Comment by j0sh77** [comment]
Looks good to me, pending test updates!

File: packages/gatsby-source-wordpress/src/utils/fetch-graphql.ts (hunk 1 above)
Comment (TylerBarnes) [reply packages/gatsby-source-wordpress/src/utils/fetch-graphql.ts:33]: Ah yep, that's right @j0sh77 ! I think let's leave it as is, it works and reads fine :)

Comment (veryspry) [reply packages/gatsby-source-wordpress/src/utils/fetch-graphql.ts:33]: Good point!

**Review by veryspry — APPROVED** [review]
This looks good to me!

**Comment by TylerBarnes** [comment]
The failing tests are unrelated and seem to be a problem with package resolution: `error Couldn't find package "scheduler@^0.20.2" required by "react-dom@^17.0.2" on the "npm" registry.`. I can confirm that the related tests are passing so I'm going to merge.

**Pull request merged**